cloud/scope/machine.go

```diff
@@ -457,6 +473,11 @@ func (m *MachineScope) InstanceSpec(log logr.Logger) *compute.Instance {
 	instance.Metadata = m.InstanceAdditionalMetadataSpec()
 	instance.ServiceAccounts = append(instance.ServiceAccounts, m.InstanceServiceAccountsSpec())
 	instance.NetworkInterfaces = append(instance.NetworkInterfaces, m.InstanceNetworkInterfaceSpec())
+	instance.GuestAccelerators = m.InstanceGuestAcceleratorsSpec()
+	if len(instance.GuestAccelerators) > 0 {
+		instance.Scheduling.OnHostMaintenance = "TERMINATE"
```

Comment:
Can we add a note to the docs on the value of `OnHostMaintenance`? I'd suggest we let users know that it will always default to `TERMINATE`.